src/main/java/com/ing/data/cassandra/jdbc/AbstractResultSet.java

```diff
@@ -58,6 +58,14 @@ abstract class AbstractResultSet implements Wrapper {
      */
     boolean isCqlType(final int columnIndex, @NonNull final DataTypeEnum type) {
         final String columnType = StringUtils.substringBefore(DataTypeEnum.cqlName(getCqlDataType(columnIndex)), "<");
+        /*
+         * BUG FIXING:
+         * isCqlType() always return false for Vector. The columnType returns looks like
+         * org.apache.cassandra.db.marshal.VectorType(14) and as such is never vector.
+         */
+        if (columnType.contains("org.apache.cassandra.db.marshal.VectorType")) {
+            return DataTypeEnum.VECTOR.equals(type);
+        }
```

Comment:
Maybe we could modify the implementation of `DataTypeEnum.cqlName(String)` to avoid duplication in `AbstractResultSet` with something like this in `DataTypeEnum`:
```java
public static String cqlName(@NonNull final com.datastax.oss.driver.api.core.type.DataType dataType) {
    final String rawCql = dataType.asCql(false, false);
    /* 
     * For vectors, dataType.asCql returns looks like 'org.apache.cassandra.db.marshal.VectorType(n)' where n is 
     * the dimension of the vector. In this specific case, return a common name not including the dimension.
     */
    if (rawCql.contains("org.apache.cassandra.db.marshal.VectorType")) {
        return VECTOR.cqlType;
    }
    return rawCql;
}
```
In this case, `DataTypeEnum.cqlName(getCqlDataType(columnIndex))` will return "Vector" and consequently `type.cqlType.equalsIgnoreCase(columnType)` will return true.

Reply:
Good call, most can be done in DataTypeEnum. I also updated `fromCqlTypeName` today.